### ISSUE TITLE
Dialog component fails "Ensures aria-hidden elements do not contain focusable elements" accessibility check

### ISSUE THREAD

**Issue by suvaso** [microsoft/fluentui#15349]
### Environment Information

- **Package version(s)**: 7.144.2
- **Browser and OS versions**: Windows 10 Version 2004 (OS Build 19041.508).  Chrome Version 85.0.4183.121 (Official Build) (32-bit).

### Describe the issue:

Dialog component threw Accessibility error when analyzed by Axe (https://www.deque.com/axe/)

### Please provide a reproduction of the issue in a codepen:

- Install Axe browser plugin from https://www.deque.com/axe/
- Go to https://fabricweb.z5.web.core.windows.net/oufr/7.144.2/#/examples/dialog
- Open the dialog
- Launch Axe from developer toolbar
- Click Analyze

#### Actual behavior:

![FluentDialog](https://user-images.githubusercontent.com/72280013/94967062-392f7280-04b3-11eb-9bb4-cf540732d21f.PNG)
eature for the various components in our documentation site.

#### Expected behavior:

No, accessibility error.

**Comment by paulgildea** [comment]
@suvaso Thanks for the issue.

@smhigley FYI on the model. This seems pretty basic did aria recently change?

**Comment by joschect** [comment]
@paulgildea I think that this appears intentional. It's a consequence of the FocusTrapZone and the way that it works to ensure that focus does not leave the area contained within. From what I can tell, it's "focusable" but that's just so it can ensure that when focus tries to leave the area via tab or other keypress that it can redirect focus inward. @JasonGore might be able to expand on this. I believe that we should have an exemption for this particular instance of this aria error but I'm not positive.

**Comment by JasonGore** [comment]
In doing [a search](https://github.com/microsoft/fluentui/issues?q=is%3Aissue+focustrapzone+aria+hidden) I found  [some duplicates](https://github.com/microsoft/fluentui/issues/14749). I also found that `aria-hidden` being applied to bumpers is a [more recent change](https://github.com/microsoft/fluentui/issues/14241) that likely triggered these issues.

**Comment by paulgildea** [comment]
@joschect @JasonGore Ok I'm confused. So is the issue with the AI tool or with the Dialog component?

**Comment by khmakoto** [comment]
Hi @suvaso, after talking with some people and looking into this I think it's safe to say that this is not an actual issue, as indicated as a possibility by the axe warning saying that it needs review.

The warning stems from our use of "bumpers" to trap focus in `FocusTrapZone`. We need to have `tabIndex="0"` in these elements because we trap focus by programmatically moving focus from these bumpers to elements inside of the `FocusTrapZone` when the bumpers are reached. However, we also need to have `aria-hidden="true"` in these elements since screen readers would attempt to read them when tabbing into them if we didn't have it.

Given that we quickly move the focus back when tabbing into the bumpers the warning that AXE throws, while in general a great rule for most scenarios, results in a non-issue for this one that should be dismissed as a non-applicable warning. If there is an actual accessibility concern from user experience regarding this and not only an automated warning then please let us know. However, if that's not the case then you should be OK resolving this as a non-issue, as we will do for the time being unless anything changes.

Thank you and let us know if there's anything else we can do to help!
